sklearn/utils/estimator_checks.py

```diff
@@ -360,55 +374,191 @@ def check_transformer_pickle(name, Transformer):
+
+    if name in ("MultinomialNB",):
+        X -= X.min()
+        test_X -= test_X.min()
+
+    estimator = Estimator()
+
+    set_random_state(estimator)
+    set_fast_parameters(estimator)
+
+    cloned = clone(estimator)
+
+    msg = ("partial_fit of the clone of estimator %s returned a different"
+           " model" % name)
+
+    classes = np.unique(y)
+    ret_val = _partial_fit(estimator, X, y, classes=classes)
+
+    # Check if the partial_fit of cloned returns the same model
+    assert_same_model(test_X, estimator,
+                      _partial_fit(cloned, X, y, classes), msg)
+
+    # Check if returns self
+    msg = "partial_fit of estimator %s did not return self" % name
+    assert ret_val is estimator, msg
+
+
+@ignore_warnings
+def check_partial_fit_reset_when_fit(name, Estimator):
+    """Check if the estimators get reset when fit is applied after a
+    partial fit.
+    """
+
+    X, y = make_blobs(n_samples=50, n_features=4, random_state=0)
+    test_X, test_y = make_blobs(n_samples=50, n_features=4, random_state=1)
+
+    if name in ("MultinomialNB",):
+        X -= X.min()
+        test_X -= test_X.min()
+
+    estimator_1 = Estimator()
+    estimator_2 = Estimator()
+
+    set_random_state(estimator_1)
+    set_fast_parameters(estimator_1)
+
+    set_random_state(estimator_2)
+    set_fast_parameters(estimator_2)
+
+    # Initial call to fit
+    _fit(estimator_1, X, y)
+    _fit(estimator_2, X, y)
+
+    # Run 2 iterations of full set of partial_fit with batches of size 10
+    classes = np.unique(y)
+    for i in range(2):
+        X_s, y_s = shuffle(X, y, random_state=i)
+        for j in range(0, 50, 10):
+            _partial_fit(estimator_2, X_s[j:10+j], y_s[j:10+j],
+                                   classes=classes)
+
+    # Test that change in features does not raise any error
+    assert_not_raises(Exception, _fit, estimator_2, X[:, :-1], y)
+
+    # Finally call fit to reset the estimator
+    _fit(estimator_2, X, y)
+
+    # Compare with the first fit model.
+    assert_same_model(test_X, estimator_1, estimator_2)
+
+
+@ignore_warnings
+def check_partial_fit_effect(name, Estimator):
+
+    X, y = make_blobs(n_samples=100, n_features=8, random_state=1)
+    # Smaller data set for 2nd partial_fit
+    dX, dy = make_blobs(n_samples=10, n_features=8, random_state=2)
+    # Test data set to compare model
+    test_X, test_y = make_blobs(n_samples=50, n_features=8, random_state=3)
+
+    if name in ("MultinomialNB",):
+        X -= X.min()
+        dX -= dX.min()
+        test_X -= test_X.min()
+
+    estimator_1 = Estimator()
+    estimator_2 = Estimator()
+
+    set_random_state(estimator_1)
+    set_fast_parameters(estimator_1)
+
+    set_random_state(estimator_2)
+    set_fast_parameters(estimator_2)
+
+    classes = np.unique(y)
+    _partial_fit(estimator_1, dX, dy, classes)
+    _partial_fit(estimator_2, X, y, classes)
+    _partial_fit(estimator_2, dX, dy, classes)
 
```

Comment:
This test could be stronger by checking that `dX` then `X` also differs from the `dX` model (showing that it's not just retained the first model), but I don't really think we need it...